test/scenarios_test.py

```diff
@@ -5,48 +5,71 @@
 
 from test_base import TestBase
 
-graph_file_path = Path(os.path.dirname(__file__)) / "data" / "12_node_ring.graphml"
 
-base = TestBase()
-base.start_server()
-print(base.warcli(f"network start {graph_file_path}"))
-base.wait_for_all_tanks_status(target="running")
-
-# Use rpc instead of warcli so we get raw JSON object
-scenarios = base.rpc("scenarios_available")
-assert len(scenarios) == 4
-
-# Start scenario
-base.warcli("scenarios run miner_std --allnodes --interval=1")
-
-
-def check_blocks():
-    # Ensure the scenario is still working
-    running = base.rpc("scenarios_list_running")
-    assert len(running) == 1
-    assert running[0]["active"]
-    assert "miner_std" in running[0]["cmd"]
-
-    count = int(base.warcli("rpc 0 getblockcount"))
-    print(f"Waiting for 30 blocks: {count}")
-    return count >= 30
-
-
-base.wait_for_predicate(check_blocks)
-
-# Stop scenario
-running = base.rpc("scenarios_list_running")
-assert len(running) == 1
-assert running[0]["active"]
-base.warcli(f"scenarios stop {running[0]['pid']}", False)
-
-
-def check_stop():
-    running = base.rpc("scenarios_list_running")
-    print(f"Waiting for scenario to stop: {running}")
-    return len(running) == 0
-
-
-base.wait_for_predicate(check_stop)
-
-base.stop_server()
+class ScenariosTest(TestBase):
+    def __init__(self):
+        super().__init__()
+        self.graph_file_path = Path(os.path.dirname(__file__)) / "data" / "12_node_ring.graphml"
+
+    def run_test(self):
+        try:
+            self.start_server()
+            self.setup_network()
+            self.test_scenarios()
+        finally:
+            self.stop_server()
+
+    def setup_network(self):
+        self.log.info("Setting up network")
+        self.log.info(self.warcli(f"network start {self.graph_file_path}"))
+        self.wait_for_all_tanks_status(target="running")
+
+    def test_scenarios(self):
+        self.check_available_scenarios()
+        self.run_and_check_scenario("miner_std")
+        self.run_and_check_scenario_from_file("src/scenarios/miner_std.py")
+
```

Comment:
e36df8ea8e1e1d16ec60373c1bf2be1601fbc1b5

heh oh kay, OR you could just put these lines in `run_test` ?

Reply:
Might add more files in the future though? (probably not, lol!)

Gunna leave though I think